applications/calculator/calculator.markup

```diff
@@ -8,15 +8,15 @@
     >
 
     <Panel padding="insets(12)" rounded>
-        <Label text="#value"/>
+        <Label id="calcscreen" text="#value"/>
     </Panel>
 
     <Container layout="grid(4, 6, 4, 4)" fill>
-        <Button icon="percent" outlined/> <Button text="CE" outlined/> <Button text="C" outlined/> <Button icon="backspace-outline" outlined/>
+        <Button icon="percent" outlined/> <Button id="clear_all" text="CE" outlined/> <Button id="clear" text="C" outlined/> <Button icon="backspace-outline" outlined/>
         <Button text="1/x" outlined/> <Button icon="exponent" outlined/> <Button icon="square-root" outlined/> <Button icon="slash-forward" outlined/>
-        <Button text="7"/> <Button text="8" /> <Button text="9" /> <Button icon="close" outlined/>
-        <Button text="4" /> <Button text="5" /> <Button text="6" /> <Button icon="minus" outlined/>
-        <Button text="1" /> <Button text="2" /> <Button text="3" /> <Button icon="plus" outlined/>
-        <Button icon="plus-minus-variant" outlined/> <Button text="0" /> <Button text="." outlined/> <Button icon="equal" filled/>
+        <Button id="7" text="7"/> <Button id="8" text="8" /> <Button id="9" text="9" /> <Button icon="close" outlined/>
```

Comment:
To ensure your code to be future proof, widgets IDs should also be valid C++ identifiers.